### PR TITLE
feat(cli): add `--docker` option to generate docker files

### PR DESCRIPTION
**Pull request by raymondfeng**
This PR adds a `docker` option for `lb4 app` cli to generate:

- Dockerfile
- .dockerignore
- Add `docker:build` and `docker:run` scripts to `package.json`

## Checklist

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated
- [ ] Agree to the CLA (Contributor License Agreement) by [clicking and signing](https://cla.strongloop.com/agreements/strongloop/loopback-next)

### CODE REVIEW

File: packages/cli/generators/app/templates/index.js.ejs [hunk 1]
```diff
@@ -6,8 +6,8 @@ if (require.main === module) {
   // Run the application
   const config = {
     rest: {
-      port: +process.env.PORT || 3000,
-      host: process.env.HOST || 'localhost',
+      port: +(process.env.PORT || 3000),
+      host: process.env.HOST,
```
Comment (raymondfeng) [inline packages/cli/generators/app/templates/index.js.ejs]: For the endpoint in the docker container to be accessed from the host operating system, the binding ip/host should be from `HOST` env var or `undefined` (all interfaces).

**Comment by emonddr** [comment]
@raymondfeng , will this PR also create the .dockerignore and Dockerfile ?

**Comment by emonddr** [comment]
@raymondfeng , using your Dockerfile and .dockerignore, and the port and host configuration you generate, I was:

- [x] able to run the lb4 app locally (not in a docker container)
- [x] able to get to successfully access the lb4 app in the docker container from my browser
- [x] able to deploy the app to Kubernetes on IBM Cloud
- [x] able to get to successfully access the lb4 app in kubernetes on IBM Cloud from my browser

**Comment by emonddr** [comment]
When running locally (outside a docker container), the output on the command line is:

```
Server is running at http://[::1]:3000
Try http://[::1]:3000/ping
```

It used to be : 

```
Server is running at http://127.0.0.1:3000
Try http://127.0.0.1:3000/ping
```
Would existing users mind the new IPv6 format ?

**Comment by emonddr** [comment]
When the app is running in the docker container and we print the logs it says:
```
Server is running at http://127.0.0.1:3000
Try http://127.0.0.1:3000/ping
```

which is good.

**Pull request merged**